### PR TITLE
fix(external_user): Refactor external_user from resource to data source.

### PR DESCRIPTION
**Pull request by Omicron7**
This resource makes more sense as a DataSource after using it.

This DataSource will still create a SemaphoreUI User if they don't exist, but it will no longer update any fields on an existing user. Previously, the resource was overwriting the `admin` and `alert` fields for existing users. This fixes that.

### CODE REVIEW

File: internal/provider/external_user_data_source_test.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		CheckDestroy:             testAccExternalUserCleanup,
+		Steps: []resource.TestStep{
+			// Read testing
+			{
+				Config: testAccExternalUserDataSourceConfigBasic(""),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.semaphoreui_external_user.test", "username", "username1"),
+					resource.TestCheckResourceAttr("data.semaphoreui_external_user.test", "name", "username1"),
+					resource.TestCheckResourceAttr("data.semaphoreui_external_user.test", "email", "username1"),
```
[the remark this answers is not in the document]
Comment (Omicron7) [reply internal/provider/external_user_data_source_test.go:69]: In this case, only `username` is provided and the the DataSource uses it as the default for `name` and `email` when they are missing. This test is checking that corner case.

**Review by cru-Luis-Rodriguez — APPROVED** [review]
I don't fully understand the change, but here some comments form a copilot chat. 

Areas for Additional Clarification:
Email Attribute:

Clarify that the email attribute should be a valid email address and not default to the username.
Provide examples in the documentation showing valid email addresses.
MarkdownDescription:

Update the MarkdownDescription to reflect that the external user is now a data source, not a resource.
External Attribute:

Ensure the external attribute is correctly set and clarify its purpose in the documentation.
Suggestions for Better Clarification:
Email Attribute:

Example: "The email attribute must be a valid email address. If not provided, it defaults to username@example.com."
MarkdownDescription:

Update to: "The external user data source allows you to lookup an external User in SemaphoreUI. This data source will create the external user in SemaphoreUI if they do not exist."
External Attribute:

Example: "The external attribute indicates if the user is linked to an external identity provider. Ensure this is set correctly during user creation."
These clarifications will help new users understand the purpose and correct usage of the attributes and the data source.
